brabbuss/rancid-tomatillos
# API and error handling

#### What does this PR do?
- adds error handling for server errors (status that are not 200-299)
- adds error page that displays if error is encountered
- add more API functionality to retrieve data needed to get path info for movie promo videos
- add promo video to movie details component
- add ReactVideo dependency 

#### How should this be manually tested?
- there is an alternate endpoint that is commented out inside the `apiCalls` file - that endpoint will reproduce a 500 status error. When swapped as the endpoint, user should see an error page with copy as well as the error code
- TDD for error handling is in the works
- should see a video for the related movie when clicking into movie details

#### Any background context you want to provide?
- created a function inside of `MovieDetails` component that will return the correct video path depending on whether user would like to see a trailer, featurette, teaser, etc. Hope to implement user facing options to choose which video to watch

#### What are the relevant tickets?
closes #28 
closes #13
